vbeam/core/probe_geometry.py

```diff
@@ -0,0 +1,109 @@
+"""A datastructure for representing a transducer probe excluding 
+the position, orientation, etc.
+"""
+
+from typing import Callable, Optional
+from vbeam.fastmath import numpy as np
+from vbeam.fastmath.traceable import traceable_dataclass
+from vbeam.util.geometry.v2 import distance
+
+identity_fn = lambda x: x  # Just return value as-is
+
+
+@traceable_dataclass(("ROC", "rx_aperture_length_s", "tx_aperture_length_s"))
+class ProbeGeometry:
+    """A container for probe parameters.
+    The aperture lengths are arc lenghts of the probe surface in azimuth and elevation. 
+    """
+    ROC: tuple[float, float] # (azimuth, elevation)
```

Comment:
We could also consider having different child `Probe` classes — like `LinearProbe` and `CurvilinearProbe` — instead of having `ROC` on probes that technically don't require it (like flat probes, in which case `ROC` would be infinity).